### PR TITLE
fix(STONEINTG-25): support index image in image-inspect task

### PR DESCRIPTION
**Pull request by MartinBasti**
Task inspect-image would report index image (multiarch image) as "From scratch" image and consequently some checks would be skipped.

This patch adds support to image index to lookup for image manifest and parse correct data.

Currently amd64 platform si expected as that's the only platform current checks are supporting.

# Before you complete this pull request ...

Look for any open pull requests in the repository with the title "e2e-tests update" and 
see if there are recent e2e-tests updates that will be applicable to your change.

### CODE REVIEW

File: task/inspect-image/0.1/inspect-image.yaml [hunk 1]
```diff
@@ -90,7 +90,29 @@ spec:
...
+            echo "[ERROR] Could not find amd64 image manifest for image $IMAGE_URL"
+            note="Task $(context.task.name) failed: Couldn't find amd64 image manifest"
+            TEST_OUTPUT=$(make_result_json -r ERROR -t "$note")
+            echo "${TEST_OUTPUT}" | tee $(results.TEST_OUTPUT.path)
+            exit 0
+          fi
+
+          # Replace image URL with new digest
+          IMAGE_URL="${IMAGE_URL/[@:]*/@$AMD64_MANIFEST_DIGEST}"
+          echo "Setting AMD64 specific image: $IMAGE_URL"
```
Comment (MartinBasti) [inline task/inspect-image/0.1/inspect-image.yaml:112]: TODO reset retries counter somehow

**Comment by MartinBasti** [comment]
Output from devtesting
```
Inspecting raw image manifest quay.io/redhat-user-workloads/sdouglas1-tenant/multi-platform-test-prod/multi-platform-test-prod:f1fcdf82edea174ffc6b52453be3858e7e0569d1 (try 1/5).
Found an index image, lookup for amd64 manifest
Setting AMD64 specific image: quay.io/redhat-user-workloads/sdouglas1-tenant/multi-platform-test-prod/multi-platform-test-prod@sha256:08de8bdfe7e3a7c8f3ed5c31b8eb51e5336cd1d594a9f30553f1ddfa7dc51a9f
Inspecting raw image manifest quay.io/redhat-user-workloads/sdouglas1-tenant/multi-platform-test-prod/multi-platform-test-prod@sha256:08de8bdfe7e3a7c8f3ed5c31b8eb51e5336cd1d594a9f30553f1ddfa7dc51a9f (try 1/5).
Image quay.io/redhat-user-workloads/sdouglas1-tenant/multi-platform-test-prod/multi-platform-test-prod@sha256:08de8bdfe7e3a7c8f3ed5c31b8eb51e5336cd1d594a9f30553f1ddfa7dc51a9f raw metadata:
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "config": {
    "mediaType": "application/vnd.oci.image.config.v1+json",
    "digest": "sha256:7280c39bbd0c83331639124996dafe666e2cd90cc50c32852d1261ba4a65068e",
    "size": 22639
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:1d5eadfc16157a6fe1c21b525fd2b7ab40e4f15e2ee26fe0e9f709462eef0991",
      "size": 79059509
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:43f725c0c0584b7187ac624f5d464f94585c82f5765ee35dd40fb83301eb6975",
      "size": 17273962
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:1754837f59411a4a3799161f6ba5c0122f70e25f84abc295e1c46637c602b581",
      "size": 197606659
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:5b119ac89323ec097c7215e35382ea1fd0022c840f7d2cd961b7f82297a4e670",
      "size": 166981655
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:4362ef4e1330155e3c1a633df8bb7d16eedbbc1bcf8a28fb886bda00a7b4a91e",
      "size": 3821663
    },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
      "digest": "sha256:5776fc20fc86a10d6ff4117c09418210100cccbf5044655903531a298d59e497",
      "size": 89505
    }
  ],
  "annotations": {
    "org.opencontainers.image.base.digest": "sha256:0200988bf4773dad494d97be5aeceb005da3b329fe6827c035509a3f6eec1ef1",
    "org.opencontainers.image.base.name": "registry.access.redhat.com/ubi9/go-toolset:1.18.9-14"
  }
}
Getting base image manifest for source image quay.io/redhat-user-workloads/sdouglas1-tenant/multi-platform-test-prod/multi-platform-test-prod@sha256:08de8bdfe7e3a7c8f3ed5c31b8eb51e5336cd1d594a9f30553f1ddfa7dc51a9f.
Detected base image: registry.access.redhat.com/ubi9/go-toolset@sha256:0200988bf4773dad494d97be5aeceb005da3b329fe6827c035509a3f6eec1ef1
Inspecting base image registry.access.redhat.com/ubi9/go-toolset@sha256:0200988bf4773dad494d97be5aeceb005da3b329fe6827c035509a3f6eec1ef1 (try 1/5).
Detected base image repository: ubi9/go-toolset
```

**Review by jsztuka — APPROVED** [review]
lgtm

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Review by dirgim — APPROVED** [review]
LGTM, just needs a rebase

**Review by kasemAlem — APPROVED** [review]

**Pull request merged**